src/main/java/org/openrewrite/maven/ResourceParser.java

```diff
@@ -90,17 +87,77 @@ public <S extends SourceFile> List<S> parseSourceFiles(
 
             long fileSize = attrs.size();
             if ((sizeThresholdMb > 0 && fileSize > sizeThresholdMb * 1024L * 1024L)) {
-                alreadyParsed.add(path);
-                logger.info("Skipping parsing " + path + " as its size + " + fileSize / (1024L * 1024L) +
+                logger.info("Parsing as Quark " + path + " as its size + " + fileSize / (1024L * 1024L) +
                         "Mb exceeds size threshold " + sizeThresholdMb + "Mb");
-                return false;
+                return true;
             }
```

Comment:
Size threshold doesn't work anymore. You return `true` here but there's no logic below to make sure that gets parsed as a quark